Assets/MixedRealityToolkit.Tools/MSBuild/Scripts/UnityProjectInfo.cs

```diff
@@ -20,107 +19,189 @@ namespace Microsoft.MixedReality.Toolkit.MSBuild
+        /// </summary>
+        private static readonly HashSet<string> ExcludedPackageReferences = new HashSet<string>()
+        {
+            "Windows.UI.Input.Spatial"
+        };
+
+        /// <summary>
+        /// Gets the name of this Unity Project.
+        /// </summary>
+        public string UnityProjectName { get; }
 
+        /// <summary>
+        /// Gets the available platforms for this Unity project.
+        /// </summary>
+        internal IEnumerable<CompilationPlatformInfo> AvailablePlatforms { get; }
+
+        /// <summary>
+        /// Gets all the parsed CSProjects for this Unity project.
+        /// </summary>
         public IReadOnlyDictionary<string, CSProjectInfo> CSProjects { get; }
 
+        /// <summary>
+        /// Gets all the parsed DLLs for this Unity project.
+        /// </summary>
         public IReadOnlyCollection<PluginAssemblyInfo> Plugins { get; }
 
         public UnityProjectInfo(IEnumerable<CompilationPlatformInfo> availablePlatforms, string projectOutputPath)
         {
-            this.availablePlatforms = availablePlatforms;
+            AvailablePlatforms = availablePlatforms;
 
-            Dictionary<string, Assembly> unityAssemblies = CompilationPipeline.GetAssemblies().ToDictionary(t => t.name);
+            UnityProjectName = Application.productName;
+
+            if (string.IsNullOrWhiteSpace(UnityProjectName))
+            {
+                UnityProjectName = "UnityProject";
+            }
 
-            Dictionary<string, CSProjectInfo> csProjects = new Dictionary<string, CSProjectInfo>();
-            CSProjects = new ReadOnlyDictionary<string, CSProjectInfo>(csProjects);
+            Plugins = new ReadOnlyCollection<PluginAssemblyInfo>(ScanForPluginDLLs());
 
-            foreach (KeyValuePair<string, Assembly> pair in unityAssemblies)
+            foreach (PluginAssemblyInfo plugin in Plugins)
             {
-                CSProjectInfo toAdd;
-                string asmDefPath = CompilationPipeline.GetAssemblyDefinitionFilePathFromAssemblyName(pair.Key);
-                if (string.IsNullOrEmpty(asmDefPath))
+                if (plugin.Type == PluginType.Native)
                 {
-                    if (!pair.Key.StartsWith("Assembly-CSharp"))
-                    {
-                        Debug.LogError($"Failed to retrieve AsmDef for script assembly: {pair.Key}");
-                    }
+                    Debug.Log($"Native plugin {plugin.ReferencePath.AbsolutePath} not yet supported for MSBuild project.");
+                }
+            }
 
-                    toAdd = new CSProjectInfo(availablePlatforms, Guid.NewGuid(), null, pair.Value, projectOutputPath);
+            CSProjects = new ReadOnlyDictionary<string, CSProjectInfo>(CreateUnityProjects(projectOutputPath));
+        }
+
+        private Dictionary<string, CSProjectInfo> CreateUnityProjects(string projectOutputPath)
+        {
+            // Not all of these will be inflated
```

Comment:
>inflated [](start = 40, length = 8)

This term is new to me, and I can't seem references to this in the rest of this file - is inflation a term for the overall process of expanding out the templates and stuff?

Reply:
Ah, inflated as in converted to a C# project, will update wording.

---
In reply to: [310753916](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5516#discussion_r310753916) [](ancestors = 310753916)